pyccel/parser/semantic.py

```diff
@@ -238,18 +239,27 @@ def annotate(self, **settings):
 
         self._semantic_done = True
 
-        #check if the ast CodeBlock is a Program to add the Garbage collecting
+        # check if the ast CodeBlock is a Program to call the Garbage collecting
         cls = (Header, EmptyNode, NewLine, Comment, CommentBlock, Module)
-        body = ast.body
-        is_module = all(isinstance(i,cls) for i in body)
-        if not isinstance(body[-1], Return) and not is_module:
-            dealloc = [Deallocate(i) for i in self._allocs]
-            body += dealloc
-            ast = CodeBlock(body)
-            self._allocs = []
+        is_module = all(isinstance(i,cls) for i in ast.body)
+        if not is_module:
```

Comment:
Is this `if` needed? If it is a module then won't `self._allocs` be empty anyway?

Reply:
i didn't test with modules or headers i put the check there to be sure that its a program I will try some tests and remove it if it's unnecessary